project_setup/scripts/iam-to-github

```diff
@@ -0,0 +1,247 @@
+        },
+    )
+    response.raise_for_status()
+
+
+def get_repo_name() -> str:
+    """Get the repository name using git."""
+    logging.debug(f"Trying to determine GitHub repository name using git.")
+    c = subprocess.run(  # nosec
+        ["git", "remote", "get-url", "origin"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+    if c.returncode != 0:
+        logging.critical("Could not determine GitHub repository name.")
+        raise Exception(c.stderr)
+    match = GIT_URL_RE.match(c.stdout.decode())
+    repo_name = match.groups()[0]  # type: ignore
+    return repo_name
+
+
+def main() -> int:
+    """Set up logging and call the requested commands."""
+    args: Dict[str, str] = docopt.docopt(__doc__, version="1.0.0")
+
+    # Validate and convert arguments as needed
+    schema: Schema = Schema(
+        {
+            "<github-personal-access-token>": And(
+                str,
+                lambda n: len(n) == 40,
+                error="<github-token> must be a 40 character personal access token.",
+            ),
+            "--log-level": And(
+                str,
+                Use(str.lower),
+                lambda n: n in ("debug", "info", "warning", "error", "critical"),
+                error="Possible values for --log-level are "
+                + "debug, info, warning, error, and critical.",
+            ),
+            "--repo": Or(
+                None,
+                And(
+                    str,
+                    lambda n: "/" in n,
+                    error='Repository names must contain a "/"',
+                ),
+            ),
+            "--suffix": Or(And(None, Use(lambda n: "")), And(str, Use(str.upper))),
+            str: object,  # Don't care about other keys, if any
+        }
+    )
+
+    try:
+        validated_args: Dict[str, Any] = schema.validate(args)
+    except SchemaError as err:
+        # Exit because one or more of the arguments were invalid
+        print(err, file=sys.stderr)
+        return 1
+
+    # Assign validated arguments to variables
+    github_token: str = validated_args["<github-personal-access-token>"]
+    log_level: str = validated_args["--log-level"]
+    repo_name: str = validated_args["--repo"]
+    secret_suffix: str = validated_args["--suffix"]
+    user_filter: str = validated_args["--user"]
+
+    # Set up logging
+    logging.basicConfig(
+        format="%(asctime)-15s %(levelname)s %(message)s", level=log_level.upper()
+    )
+
+    # If the user does not provide a repo name we'll try to determine it from git
+    if not repo_name:
+        repo_name = get_repo_name()
+    logging.info(f"Using GitHub repository name: {repo_name}")
+
+    if user_filter:
+        logging.info(f"Filter to users containing: {user_filter}")
+
+    if secret_suffix:
+        if not secret_suffix.startswith("_"):
+            secret_suffix = f"_{secret_suffix}"
+        logging.info(f'Apending "{secret_suffix}" to secret names.')
```

Comment:
Typo: Should be "Appending", not "Apending".

Reply:
Fixed in:  c255822